lib/lob/util.ex

```diff
@@ -46,6 +46,14 @@ defmodule Lob.Util do
   end
 
   @spec transform_argument({any, any}) :: list
+  defp transform_argument({:merge_variables, v}) do
+    [{"merge_variables", Poison.encode!(v)}]
```

Comment:
Adding these function clauses because with nested structs we would try to convert the nested struct to a string in this function clause and it would error:
```elixir
  defp transform_argument({k, v}) when is_map(v) do
    Enum.map(v, fn {sub_k, sub_v} ->
      {"#{to_string(k)}[#{to_string(sub_k)}]", to_string(sub_v)}
    end)
  end
```

Reply:
I could always encode to JSON if the value is a map but we also use this function when we build the query string so just decided to only do this for merge variables

Reply:
What was the error for the above? And won't we need to make this recursive now for nested nested objects or no?

Reply:
`Lob.Util.build_body(%{merge_variables: %{money: %{test: "test"}}})` throws this error:
```
** (Protocol.UndefinedError) protocol String.Chars not implemented for %{test: "test"}. This protocol is implemented for: Float, DateTime, Time, List, Version.Requirement, Atom, Integer, Version, Date, BitString, NaiveDateTime, URI
```
The function clauses pattern matching on the `merge_variable` key are just converting the nested structs into JSON before they run into this error, which is prob not the best solution. Looking into a recursive solution right now.

Reply:
Update: spent some time yesterday trying to implement a recursive solution and there were some issues nesting lists inside maps. For example:

```
{"merge_variables[data][test][tester][name][0][yo]", "WHAT UP"}
```

Would be encoded into this JSON:

```JSON
{
  "data": {
    "test": {
      "tester": {
        "name": [
          {
            "[yo]": "WHAT UP"
          }
        ]
      }
    }
  }
}
```

It's unclear to me why the brackets are being added to surround the `yo` key and I don't think it's worth much more effort investigating.

I think this solution of pattern matching `merge_variables` and setting the content-type explicitly to JSON just for these fields is the best option.